velox/exec/TaskDriverOperatorLifecycle.md

```diff
@@ -0,0 +1,96 @@
+# Task, Driver, Operator lifecycle
+
+__TL;DR__ Task outlives Drivers and Operators. There are circular references between
+Tasks and Drivers which are cleared when Task drops references to Drivers.
+Task::terminate contains catch-all logic to clear things up in case of abnormal
+or early termination.
+
+## PartitionedOutputBufferManager
+
+PartitionedOutputBuffer holds a reference to the task. This reference is used to
+notify the task that all output has been consumed by calling
+task->setAllOutputConsumed() from PartitionedOutputBuffer::deleteResults.
+
+This reference is dropped when PartitionedOutputBuffer is destroyed which
+happens in (1) PartitionedOutputBufferManager::deleteResults (happy path) and
+(2) PartitionedOutputBufferManager::removeTask (error path).
+
+In Prestissimo application, PartitionedOutputBufferManager::deleteResults is called by
```

Comment:
Nit : It would really help to write a comment in this presto_cpp code referencing this doc instead of stating it here as that requires a context switch.